tests/slack_sdk/web/test_web_client_issue_921_custom_logger.py

```diff
@@ -0,0 +1,35 @@
+import unittest
+from logging import Logger
+
+from slack_sdk.web import WebClient
+from tests.slack_sdk.web.mock_web_api_server import (
+    setup_mock_web_api_server,
+    cleanup_mock_web_api_server,
+)
+
+
+class TestWebClient_Issue_921_CustomLogger(unittest.TestCase):
+    def setUp(self):
+        setup_mock_web_api_server(self)
+
+    def tearDown(self):
+        cleanup_mock_web_api_server(self)
+
+    def test_if_it_uses_custom_logger(self):
+        logger = CustomLogger("test-logger")
+        client = WebClient(
+            base_url="http://localhost:8888", token="xoxb-api_test", logger=logger,
+        )
+        client.chat_postMessage(channel="C111", text="hello")
+        self.assertTrue(logger.called)
+
+
+class CustomLogger(Logger):
+    called: bool
+
+    def __init__(self, name, level="DEBUG"):
+        Logger.__init__(self, name, level)
+        self.called = False
+
+    def debug(self, msg, *args, **kwargs):
+        self.called = True
```

Comment:
🤔 Would it be worthwhile to refactor `CustomLogger` to be imported from `tests.slack_sdk` since it's repeated a few times? I don't think it should block this PR landing but it's something to consider in the future.

Reply:
Thanks. I intentionally avoided sharing a class across async and sync tests for complete separation. But as you pointed out, we may want to use a single utility for both web clients and webhook clients. Let me go with this approach this time but I am open to improve this in the future.